### PR TITLE
Change deflection angle jacobian to lens equation jacobian

### PR DESCRIPTION
**Pull request by ConnorStoneAstro**

### CODE REVIEW

File: test/test_sie.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def test_jacobian():
     # Parameters
     z_s = torch.tensor(1.2)
     x = torch.tensor([0.5, 0.912, -0.442, 0.7, pi / 3, 1.4])
-    J = lens.jacobian_reduced_deflection_angle(thx, thy, z_s, lens.pack(x))
+    J = lens.jacobian_lens_equation(thx, thy, z_s, lens.pack(x))
```
Comment (AlexandreAdam) [inline test/test_sie.py:48]: Do we have a test comparing the finite difference and the autograd method? At least adding a check that we can call the method and that it works woud be appropriate.

**Review by AlexandreAdam — CHANGES_REQUESTED** [review]
This looks very good! After the unit test this pull request will be ready to merge.

File: test/test_sie.py (hunk 1 above)
Comment (ConnorStoneAstro) [reply test/test_sie.py]: Good idea! Adding that now

**Review by AlexandreAdam — APPROVED** [review]
Great work!

**Pull request merged**